components/model-selector.tsx

```diff
@@ -112,134 +112,189 @@ export function ModelSelector({
         ? `${selectedModel.modelId} ${dict.modelConfig.clickToChange}`
         : `${dict.modelConfig.usingServerDefault} ${dict.modelConfig.clickToChange}`
 
+    const wrapperRef = useRef<HTMLDivElement | null>(null)
+    const [showLabel, setShowLabel] = useState(true)
+
+    // Threshold (px) under which we hide the label (tweak as needed)
+    const HIDE_THRESHOLD = 250
+    const SHOW_THRESHOLD = 250
```

Comment:
The HIDE_THRESHOLD and SHOW_THRESHOLD are both 250, which defeats the hysteresis mechanism. At exactly 250px width, the state will toggle infinitely

```suggestion
    const HIDE_THRESHOLD = 240
    const SHOW_THRESHOLD = 260
```